mariadb/templates/mariadb-deployment.yaml

```diff
@@ -1,40 +1,36 @@
-apiVersion: v1
-kind: ReplicationController
+apiVersion: extensions/v1beta1
+kind: Deployment
 metadata:
-  name: {{ printf "%s-%s" .Release.Name "mariadb" | trunc 24 }}
+  name: {{ template "fullname" . }}
   labels:
```

Comment:
I'm adding the `chart` label. The idea behind the `version` label was to provide for rolling updates. Not sure if it's required anymore. Removing it for now.

W.r.t to the `provider` label, it's the primary `selector` label defined on the mariadb service. I don't think we can replace it with the `chart` label.